File: sd/python/sd_openai_sdk.py
import os
from openai import OpenAI
import uuid

# Function to generate a unique job ID
def generate_job_id():
    return "sdk_image_" + str(uuid.uuid4())

# Main function to interact with OpenAI API using SDK
def main():
    api_key = "your_user_id#your_api_key" 
    base_url = "https://llm-gateway.heurist.xyz"

    # Initialize OpenAI client with API key and base URL
    client = OpenAI(api_key=api_key, base_url=base_url)

    # Define the model ID and job parameters
    model_id = 'FLUX.1-dev'
    payload = {
        "job_id": generate_job_id(),
        "model_input": {
            "SD": {
                "prompt": "A beautiful landscape with mountains and a river",
                "neg_prompt": "Avoid any signs of human presence",
                "num_iterations": 20,
                "width": 1024,
                "height": 512,
                "guidance_scale": 7.5,
                "seed": -1
            }},
        "model_id": model_id,
        "deadline": 60,
        "priority": 1
    }

    try:
        # Submit the job using the OpenAI SDK (this is the equivalent of sending the job via REST API)
        response = client.jobs.create(
            model=model_id,
            input=payload
        )

        # Handle the response and print the relevant output
        print(f"Generated job ID: {payload['job_id']}")
        print(f"Response status: {response.status_code}")
        
        if response.status_code == 200:
            print('Job submitted successfully')
        else:
            print(f"Error submitting job: {response.text}")
        
        print(f"Generated image URL: {response.text}")

    except Exception as e:
        print(f"An error occurred: {e}")

if __name__ == "__main__":
    main()

File: sd/js/sd_openai_sdk.js
const { v4: uuidv4 } = require('uuid');
const { OpenAI } = require('openai'); 

function generateJobId() {
    return "sdk_image_" + uuidv4();
}

async function main() {
    console.log('Starting image generation job...');

    // Initialize OpenAI client with the provided API key
    const client = new OpenAI({
        apiKey: "your_user_id#your_api_key",
        baseURL: 'https://llm-gateway.heurist.xyz',  // Make sure to adjust the URL if necessary
    });

    const payload = {
        "job_id": generateJobId(),
        "model_input": {
            "SD": {
                "prompt": "A beautiful landscape with mountains and a river",
                "neg_prompt": "Avoid any signs of human presence",
                "num_iterations": 20,
                "width": 1024,
                "height": 512,
                "guidance_scale": 7.5,
                "seed": -1
            }
        },
        "model_id": "HeuristLogo",  // Make sure to use the correct model ID for image generation
        "deadline": 60,
        "priority": 1
    };

    console.log('Job ID:', payload.job_id);
    console.log('Submitting image generation job with OpenAI SDK...');

    try {
        // Call OpenAI SDK method for image generation (adjust for actual API endpoint)
        const response = await client.images.create({
            prompt: payload.model_input.SD.prompt,
            negative_prompt: payload.model_input.SD.neg_prompt,
            num_iterations: payload.model_input.SD.num_iterations,
            width: payload.model_input.SD.width,
            height: payload.model_input.SD.height,
            guidance_scale: payload.model_input.SD.guidance_scale,
            seed: payload.model_input.SD.seed,
        });

        // Assuming the response contains the image URL or image data
        console.log('Image generation successful. Response:', response);
        return response;
    } catch (error) {
        console.error('Failed to generate image:', error.message);
        throw error;
    }
}

if (require.main === module) {
    main().catch(console.error);
}
